examples/megacli.cpp

```diff
@@ -2500,6 +2500,35 @@ void exec_treecompare(autocomplete::ACState& s);
 void exec_querytransferquota(autocomplete::ACState& s);
 
 
+void exec_showattributes(autocomplete::ACState& s)
+{
+    if (Node* n = nodeFromRemotePath(s.words[1].s))
+    {
+        for (auto pair : n->attrs.map)
+        {
+            char namebuf[10]{};
+            AttrMap::nameid2string(pair.first, namebuf);
+            if (pair.first == 'c')
+            {
+                FileFingerprint f;
+                f.unserializefingerprint(&pair.second);
+                cout << namebuf << ": " << pair.second << " (fingerprint: size " << f.size << " mtime " << f.mtime
+                    << " crc " << std::hex << f.crc[0] << " " << f.crc[1] << " " << f.crc[2] << " " << f.crc[3] << std::dec << ")"
```

Comment:
Can you output crc in a loop so we don't need to know how many elements it holds?

Reply:
I think it's a lot more compact this way - with the rules around curly braces the function would be at least 8 lines longer if those were loops.   The crc array is fundamental and won't be changing.  cheers

Reply:
I like that you can see the future Matt :)